### DISCUSSION TITLE
Inconsistency in Beta Value in runScripts_v2.py for FieldInversion Task

### DISCUSSION THREAD

**Discussion by lnhang0618** [mdolab/dafoam#595] [Bug] [answered]
Dear Professor He,

I am facing an uncertain issue in parallel computing. I used runScripts_v2.py for the FieldInversion task. To my surprise, while running the runScripts_v2 version, I noticed that the beta_fieldInversion term in primalSolver is not zero, which seems illogical. On further investigation, I identified the problem here.

In this code section, I found that the values of dvDict and DV_listGlobal are different, leading to a beta value, which should be 1, being set to 0 at every alternate position in the list. This error only occurs under the "--task opt" setting. I didn't encounter this issue while running runPrimal and runAdjoint. I have attempted to debug this issue but am struggling to determine where dvDict is initialized and setDesignVars is called in the "opt" mode.

Best regards,
Luo yinhang
![1708855185253](https://github.com/mdolab/dafoam/assets/77599661/82f39f8d-0090-497f-8fbd-098e53e27316)
![1708855648027](https://github.com/mdolab/dafoam/assets/77599661/9bb9f00b-f5d1-4cb8-bc4c-87b03136886c)
![1708856106869](https://github.com/mdolab/dafoam/assets/77599661/36d3c1c4-aacc-45ba-8057-244c029c28e8)

**Comment by lnhang0618** [comment]
In my tests, runScript.py does not exhibit this issue, but it has an additional bug that hinders parallel processing. While runScript_v2.py is free from the parallel bug, the beta's dvDict seems to have some problems.

**Comment by lnhang0618** [comment]
I have successfully identified where the problem occurs.But further debugging is somewhat beyond my capabilities.
![1708861907709](https://github.com/mdolab/dafoam/assets/77599661/8d4e6c0f-f07c-463f-8218-0f05d5483ac4)
![1708861933824](https://github.com/mdolab/dafoam/assets/77599661/8c0a6d21-41cf-4410-a3ce-0f4bf93f59b7)

**Comment by friedenhe** [comment]
This is strange. Can you share your case (please minimize its size) so that we can reproduce the issue?

**Reply by lnhang0618** [reply]
[S809.zip](https://github.com/mdolab/dafoam/files/14407973/S809.zip)
Certainly, Here is the case！

**Comment by friedenhe** [comment, marked as answer]
We reproduced the issue you had. It is likely to be caused by the mix of real and complex numbers in DVGeo. Instead of digging into it, I suggest you try our new API for setting non-geometry design variables (there is no need to set DVGeo, USMesh, DVCon, etc). See attached for a runScript template. It should fix the issue you have. 

[runScript_v2.py.zip](https://github.com/mdolab/dafoam/files/14423701/runScript_v2.py.zip)

**Reply by 15175829310** [reply]
Don't you need an FFD box in there?
![22222](https://github.com/mdolab/dafoam/assets/102606153/c85f44fe-3a51-4b94-9481-c7e88ab64101)

**Reply by lnhang0618** [reply]
I think FFD is not necessary in FieldInversion since it is no need to optimize geometry

**Reply by 15175829310** [reply]
Thanks for your reply, I am testing this script
